aws/amazon-ecs-ami
# [AL2022] Refactor ECS optimized AL2022 AMI building scripts - Part 1

### Summary
This PR
* Pull the latest changes from the main branch
* Rename variables
* Add `most_recent = true` to al2022.pkr.hcl and al2022arm.pkr.hcl to ensure the latest source AMIs are used
* Update AL2022 release server to get efs package
* Reorder entries in the generate-release-vars.sh

### Implementation details
1. CHANGELOG.md - merge from upstream main branch (commit 8368462).
2. al1.pkr.hcl,  al2.pkr.hcl, al2arm.pkr.hcl, al2gpu.pkr.hcl, al2inf.pkr.hcl and release.auto.pkrvars.hcl - merge from upstream main branch (commit c1ffc2b).
3. al2022.pkr.hcl - add `most_recent = true`, reorder provisioners to run workaround script before installing additional packages for now, uncomment  the provisioner to install available packages and add `"ECS_INIT_URL=${var.ecs_init_url_al2022}"` for  the provisioner to execute install-ecs-init.sh.
4. al2022arm.pkr.hcl - add `most_recent = true and fix os_version. 
5. generate-release-vars.sh - reorder entries and rename variables for AL2 x86 and arm AMI. The release.auto.pkrvars.hcl will look like following in the future.
```
ami_version          = "20220209"
ecs_agent_version    = "v1.59.0"
ecs_init_rev         = "1"
docker_version       = "20.10.7"
containerd_version   = "1.4.6"
source_ami_al1       = "amzn-ami-minimal-hvm-2018.03.0.20220209.0-x86_64-ebs"
source_ami_al2       = "amzn2-ami-minimal-hvm-2.0.20220207.1-x86_64-ebs"
source_ami_al2arm    = "amzn2-ami-minimal-hvm-2.0.20220207.1-arm64-ebs"
source_ami_al2022    = "al2022-ami-minimal-2022.0.20220204.0-kernel-5.10-x86_64"
source_ami_al2022arm = "al2022-ami-minimal-2022.0.20220204.0-kernel-5.10-arm64"
```

4. scripts/al2022/install-workarounds.sh - update release server to `2022.0.20220204` to get amazon-efs-utils build from internal Koji repo and remove EFS install block
5. variables.pkr.hcl - rename packages to `packages_al2`, update `packages_al2022 = "amazon-efs-utils"`and add the `ecs_init_url_al2022` block

### Testing
1. Built AL2 AMI successfully, and validated with the TestSimpleExit.
2. Built AL2022 AMI successfully

New tests cover the changes: no

### Description for the changelog
N/A

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
